src/System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs

```diff
@@ -400,6 +400,45 @@ public static Span<T> AsSpan<T>(this T[] array, int start)
             return new Span<T>(ref Unsafe.Add(ref Unsafe.As<byte, T>(ref array.GetRawSzArrayData()), start), array.Length - start);
         }
 
+        /// <summary>
+        /// Creates a new span over the portion of the target array.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Span<T> AsSpan<T>(this T[] array, Index startIndex)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
```

Comment:
The existing overloads allow array to be null if startIndex is 0.  Should the same be done here for consistency?

Reply:
I'll do that